us_election_data_quality_check_and_fixes.sql

```diff
@@ -0,0 +1,206 @@
+----------------------------------------------------------------------------------
+--QUALITY CHECK TABLICY COUNTY_FACTS
+----------------------------------------------------------------------------------
+
+--sprawdzenie, czy fips zawsze dodatni
+select * from county_facts cf 
+where fips < 0 
+or fips isnull ;
+
+--sprawdzenie, czy skr�t sk�ada si� zawsze z dw�ch liter
+select 
+	count(*)
+,	case	when upper(state_abbreviation) similar to '[A-Z]{2}' then 'valid state abbr'
+			else 'invalid state abbr' end state_abbr_check
+from county_facts cf 
+group by state_abbr_check ;
+
+--^52 invalid state abbreviation --> 51 stan�w + USA
+select 
+	area_name 
+,	state_abbreviation 
+,	case	when upper(state_abbreviation) similar to '[A-Z]{2}' then 'valid state abbr'
+			else 'invalid state abbr' end state_abbr_check
+from county_facts cf 
+order by state_abbr_check, area_name 
+limit 52 ;
+
+--sprawdzenie, czy wszystkie hrabstwa maj� wpisany state_abbreviation 
+--(oraz czy wszystkie stany maj� null w state_abbreviation) 
+select 
+	fips
+,	area_name 
+,	state_abbreviation 
+from county_facts cf 
+where state_abbreviation isnull
+order by fips ;
+
+select count(*)
+from county_facts cf 
+where state_abbreviation isnull ;
+
+select 
+	fips
+,	area_name 
+,	state_abbreviation 
+from county_facts cf 
+where state_abbreviation is not null 
+and state_abbreviation not similar to '[A-Z]{2}'
+order by fips ;
+
+--^USA i Alabama nie maj� przypisanej warto�ci null
+--dla ujednolicenia przypisuj� null tym polom 
+update county_facts set state_abbreviation = null 
```

Comment:
To jest jak najbardziej dobry kierunek, ale gdy robimy updaty w tabelkach, z których korzysta ktoś więcej niż Ty, warto to skonsultować z pozostałymi, żeby zapewnić jednolitość danych dla wszystkich użytkowników - takiego updata warto umieścić np w osobnym skrypcie z innymi updatami, a nie w skrypcie, który zawiera też analizę. Do przedyskutowania, czy chcemy to stąd wyprowadzić, czy zostawić tak jak jest, bo np. zakładamy, że ten skrypt będzie jedynym, w którym takie updaty będą się pojawiać. Zazwyczaj jednak bardziej pożądana jest pierwsza opcja, bo trzymanie takich poleceń w osobnym skrypcie ułatwia w razie czego szybkie sprawdzenie, jakie modyfikacje były dokonywane w danych.